app/internal/liability/liability.go

```diff
@@ -93,132 +82,76 @@ func (l *Liability) AddRevertLiability(asset Asset, amount decimal.Decimal) erro
 	return nil
 }
 
-// Add combines 2 liabilities.
-// 1: REQ: { BTC: 12, ETH: 14 }, ACK: {RTC: 2}
-// 2: REQ: { BTC: 23, GOLD: 25 }, ACK: {USDT: 200}
-// Result: REQ: { BTC: 35, ETH: 14, GOLD: 25 }, ACK: {RTC: 2, USDT: 200}}
-func (l *Liability) Add(newLiability *Liability) {
-	for lAsset, lAmount := range newLiability.ACK {
-		if val, ok := l.ACK[lAsset]; ok {
-			l.ACK[lAsset] = val.Add(lAmount)
-		} else {
-			l.ACK[lAsset] = lAmount
-		}
+// validate validates given params for further operations.
+func (l *Liabilities) validate(asset Asset, amount decimal.Decimal) error {
+	if _, ok := l.REQ[asset]; !ok {
+		return ErrNoReqLiability
 	}
 
-	for lAsset, lAmount := range newLiability.REQ {
-		if val, ok := l.REQ[lAsset]; ok {
-			l.REQ[lAsset] = val.Add(lAmount)
-		} else {
-			l.REQ[lAsset] = lAmount
-		}
+	if amount.Cmp(l.REQ[asset]) == 1 {
+		return ErrInvalidOperation
 	}
-}
-
-// NewLiabilitiesMap creates new liabilityMap from existing liability.
-func NewLiabilitiesMap(index uint, liability *Liability) *LiabilitiesMap {
-	liabilitiesMap := make(map[uint]*Liability)
-	liabilitiesMap[index] = liability
 
-	return &LiabilitiesMap{
-		Liabilities: liabilitiesMap,
-	}
+	return nil
 }
 
-// Add combines liabilityMap with already existing one.
-func (lm *LiabilitiesMap) Add(liabilitiesMap *LiabilitiesMap) {
-	for index, liability := range liabilitiesMap.Liabilities {
-		if lm.Liabilities[index] == nil {
-			lm.Liabilities[index] = liability
-		} else {
-			lm.Liabilities[index].Add(liability)
-		}
+// AddRequestLiability adds new request liability to liabilities state.
+func (ls LiabilitiesState) AddRequestLiability(from, to uint, asset Asset, amount decimal.Decimal) {
+	liabilitiesMap, found := ls[from]
+	if !found {
+		ls[from] = make(LiabilitiesMap)
 	}
-}
-
-// NewLiabilityState creates new liabilityState.
-func NewLiabilityState() *LiabilityState {
-	state := make(map[uint]*LiabilitiesMap)
 
-	return &LiabilityState{
-		State: state,
+	_, found = liabilitiesMap[to]
+	if !found {
+		ls[from][to] = NewLiabilities()
 	}
-}
 
-// AddLiability combines existing liabilities map with input liability.
-func (ls *LiabilityState) AddLiability(from, to uint, liability *Liability) {
-	lMap := NewLiabilitiesMap(to, liability)
-	if ls.State[from] == nil {
-		ls.State[from] = lMap
-	} else {
-		ls.State[from].Add(lMap)
-	}
+	ls[from][to].AddRequestLiability(asset, amount)
 }
 
-// AddRequestLiability adds new request liability to liability state.
-func (ls *LiabilityState) AddRequestLiability(from, to uint, asset Asset, amount decimal.Decimal) error {
-	liability := NewLiability()
-	liability.AddRequestLiability(asset, amount)
-	ls.AddLiability(from, to, liability)
-
-	return nil
-}
+// AddAcknowledgeLiability adds new acknowledge liability to liabilities state.
+func (ls LiabilitiesState) AddAcknowledgeLiability(from, to uint, asset Asset, amount decimal.Decimal) error {
+	_, found := ls[from][to]
```

Comment:
here, there is no reason to execute sth if there are no `ls[from][to]`
it means that map is empty, and it 100% there is no `req` type which we need to do `revert`/`acknowledge`